lib/proxy/client.go

```diff
@@ -463,6 +488,7 @@ func (c *Client) getConnections(proxyIDs []string) ([]*clientConn, bool, error)
 
 	ids := make(map[string]struct{})
 	var conns []*clientConn
+	defer c.config.connShuffler(conns)
```

Comment:
This is equivalent to
```go
	defer c.config.connShuffler(nil)
```
because arguments to deferred calls are evaluated at the point of defer (see https://go.dev/play/p/Jdg51E-uc-T)

A fix could be 
```suggestion
	defer func() { c.config.connShuffler(conns) }()
```
but I feel that it would be easier to just call the shuffler before each of the two returns.

Reply:
Good catch. Did not know that :smile: